examples/Makefile-nrf52840

```diff
@@ -87,14 +87,27 @@ CONFIG_FILE_PATH = $(AbsTopSourceDir)/examples/platforms/nrf52840/
 COMMONCFLAGS                   := \
     -fdata-sections               \
     -ffunction-sections           \
-    -Os                           \
-    -g                            \
     -D$(CONFIG_FILE)              \
     -I$(CONFIG_FILE_PATH)         \
     $(NULL)
 
 include $(dir $(abspath $(lastword $(MAKEFILE_LIST))))/common-switches.mk
 
+ifeq ($(DEBUG),1)
+configure_OPTIONS += --enable-optimizations=no
+COMMONCFLAGS                   += \
+    -O0                           \
```

Comment:
nlbuild-autotools already attempts to handle `-O*` and `-g*` flags. I suggest relying on nlbuild-autotools to apply the flags and make necessary updates to nlbuild-autotools to support `-ggdb3`. Doing so would make this generally useful for all users of nlbuild-autotools.